src/header/raw.rs

```diff
@@ -1,229 +1,163 @@
-use std::borrow::Cow;
 use std::fmt;
+use http::buf::{self, MemSlice};
 
 /// A raw header value.
 #[derive(Clone, PartialEq, Eq)]
-pub struct Raw(Lines);
+pub struct Raw(MemSlice);
```

Comment:
I was actually thinking more that internally, we'd have `Line` change from `Cow` to another enum:

```rust
enum Line {
    Static(&'static [u8]),
    Owned(Vec<u8>),
    Shared(MemSlice),
}
```